js/data/bucket/symbol_bucket.js

```diff
@@ -358,7 +366,7 @@ class SymbolBucket {
             // the buffers for both tiles and clipped to tile boundaries at draw time.
             const addToBuffers = inside || mayOverlap;
             this.addSymbolInstance(anchor, line, shapedTextOrientations, shapedIcon, this.layers[0],
-                addToBuffers, this.symbolInstancesArray.length, this.collisionBoxArray, feature.index, feature.sourceLayerIndex, this.index,
```

Comment:
Per conversation w @mollymerp, I replaced the (unused) `index` field with a `featureIndex`, which provides a mapping back to from the symbol instance to the feature it came from.  This allows us to retrieve the feature's properties down in `place()`.

This will likely need a bit of reworking when we return to the cross-source symbol placement quest, but since we don't know exactly what that'll look like, I think we should cross that bridge when we come to it.